File: problems/d/dicecup/dicecup.py
#! /usr/bin/python3
"""
Author: Rudi César Comiotto Modena
Email: rudi.modena@gmail.com
"""

def main():
    n, m = [int(val) for val in input().split()]

    results = [0 for _ in (range(1, (n + m) + 2))]
    for outcome_a in range(1, n + 1):
        for outcome_b in range(1, m + 1):
            total = outcome_a + outcome_b
            results[total] += 1

    max_outcomes = max(results)
    for number, total in enumerate(results):
        if total == max_outcomes:
            print(number)


if __name__ == '__main__':
    main()

File: problems/d/dicecup/dicecup.cpp
/*
Author: Rudi César Comiotto Modena
Email: rudi.modena@gmail.com
*/

#include <bits/stdc++.h>

using namespace std;

int main() {
    int n, m;

    cin >> n >> m;

    vector<int> results;
    for (int i=1; i< n + m + 2; i++) {
        results.push_back(0);
    }

    for (int i=1; i<n+1; i++) {
        for (int j=1; j<m+1; j++) {
            int total = i + j;
            results[total]++;
        }
    }

    int max_outcome = *max_element(begin(results), end(results));
    for (size_t i=0; i<results.size(); i++) {
        if (results[i] == max_outcome){
            cout << i << endl;
        }
    }
}